### DISCUSSION TITLE
What will happen "Suspicious requests to set or unset"?

### DISCUSSION THREAD

**Discussion by longlong10086** [spack/spack#25884] [Q&A]
When installing the self-defined software, the environment variable does not take effect, I  found "Warning: Suspicious requests to set or unset 'MY_VAR' found" . What are the possible reasons for this situation?

**Comment by alalazo** [comment]
We should probably make the warning message a bit more clear. What it means is that, collecting all the requests to modify the environment variables for that software, there are conflicting requests to:
1. Set or modify MY_VAR
2. Unset it
